test/unit/spec/IDPDiscovery_spec.js

```diff
@@ -560,7 +560,8 @@ function (Q, OktaAuth, WidgetUtil, Okta, Util, AuthContainer, IDPDiscoveryForm,
           deferred.resolve('thisIsTheDeviceFingerprint');
           return deferred.promise;
         });
-        return setup({ features: { securityImage: true, deviceFingerprinting: true }})
+        return setup({ features: { securityImage: true, deviceFingerprinting: true,
```

Comment:
we have tests for 
- `deviceFingerprinting: true` and `useDeviceFingerprintForSecurityImage: true`
- `deviceFingerprinting: false` and `useDeviceFingerprintForSecurityImage: false`

I think we should add one more here for 
- `deviceFingerprinting: true` and `useDeviceFingerprintForSecurityImage: false`

that way we cover the 3 scenarios in this test (similar to the other test you are changing)

Reply:
what about     
* `deviceFingerprinting: false` and `useDeviceFingerprintForSecurityImage: true`

Reply:
@mauriciocastillosilva-okta  @haishengwu-okta  I have the same cases tested in Primary_auth spec. 
Dint want to repeat all the flows for idp since it would go through the same flow. 

https://github.com/okta/okta-signin-widget/pull/629/files#diff-dc06304da4eb7ee5361d32a9f5850bb0R973

Reply:
IdpDiscoveryForm extends from PrimaryAuthForm, so you have a point. We already have tests that are somehow duplicated for both specs, which is not ideal. Probably we should look into refactoring these tests in the future to remove all the dups.
But with the current state of things, it won't hurt having the tests in both specs...